plugins/field-numpad/src/field_numpad.ts

```diff
@@ -0,0 +1,243 @@
+ * https://github.com/google/blockly/blob/master/core/field.ts
+ */
+type FieldNumpadValidator = Function;
+/* eslint-enable @typescript-eslint/ban-types */
+
+/**
+ * Numpad field.
+ */
+export class FieldNumpad extends Blockly.FieldNumber {
+  /**
+   * Class for an number numpad field.
+   * @param value The initial value of the field. Should
+   *    cast to a number. Defaults to 0.
+   * @param min Minimum value.
+   * @param max Maximum value.
+   * @param precision Precision for value.
+   * @param validator A function that is called to validate
+   *    changes to the field's value. Takes in a number & returns a validated
+   *    number, or null to abort the change.`
+   * @param config A map of options used to configure the field.
+   *    See the [field creation documentation]{@link
+   * https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/number#creation}
+   *    for a list of properties this parameter supports.
+   */
+  constructor(value?: string | number, min?: string | number,
+      max?: string | number, precision?: string | number,
+      validator?: FieldNumpadValidator, config?: FieldNumpadConfig) {
+    super(value, min, max, precision, validator, config);
+  }
+
+  /**
+   * Constructs a FieldNumpad from a JSON arg object.
+   * @param options A JSON object with options (value, min, max, and
+   *                          precision).
+   * @return The new field instance.
+   * @package
+   * @nocollapse
+   */
+  static fromJson(options: FieldNumpadOptions): FieldNumpad {
+    return new FieldNumpad(
+        options.value, undefined, undefined, undefined, undefined, options);
+  }
+
+  /* eslint-disable @typescript-eslint/naming-convention */
+  /**
+   * Show the inline free-text editor on top of the text along with the numpad
+   *    editor.
+   * @param e Optional mouse event that triggered the field to
+   *     open, or undefined if triggered programmatically.
+   * @param quietInput Quiet input (prevent focusing on the editor).
+   */
+  protected showEditor_(e?: Event, quietInput?: boolean) {
+    // Always quiet the input for the super constructor, as we don't want to
+    // focus on the text field, and we don't want to display the modal
+    // editor on mobile devices.
+    super.showEditor_(e, true);
+
+    // Build the DOM.
+    const editor = this.dropdownCreate_();
+
+    Blockly.DropDownDiv.getContentDiv().appendChild(editor);
+
+    const sourceBlock = this.getSourceBlock() as Blockly.BlockSvg;
+
+    const primary = sourceBlock.getColour() || '';
+    const tertiary = sourceBlock.getColourTertiary() || '';
+    Blockly.DropDownDiv.setColour(primary, tertiary);
+
+    Blockly.DropDownDiv.showPositionedByField(
+        this, undefined);
+  }
+
+  /**
+   * Updates the numpad when the field rerenders.
+   */
+  protected render_() {
+    super.render_();
+  }
+
+  /**
+   * Creates a table cell in the provided table row
```

Comment:
Can we get a period at the end of the function descriptions and each parameter line?  Here and several places below.